### PR TITLE
Updating change log for release 2.4.0-alpha.1.

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Updating change log for release 2.4.0-alpha.1.

### CODE REVIEW

**Comment by Zayankovsky** [comment]
Could you fix the changelog line for #5490? After merging the PR it ended up in the section for version `2.3.0-rc.2`.

**Comment by cafesilencio** [comment]
> Could you fix the changelog line for #5490? After merging the PR it ended up in the section for version `2.3.0-rc.2`.

Should that be in the change log section for 2.4.0?

**Comment by Zayankovsky** [comment]
> > Could you fix the changelog line for #5490? After merging the PR it ended up in the section for version `2.3.0-rc.2`.
> 
> Should that be in the change log section for 2.4.0?

Yes, it should have been in the `Unreleased` section, so you can move it to `2.4.0-alpha.1` now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,14 +4,29 @@ Mapbox welcomes participation and contributions from everyone.
...
   Most of the Navigation SDK APIs that rely only on `DirectionsRoute` are now marked as `Deprecated` since there might be features introduced in the future that would require the `NavigationRoute` instances instead.
   There are `NavigationRoute#toDirectionsRoute()` and `DirectionsRoute#toNavigationRoute()` compatibility extensions available, however, the latter is lossy since `DirectionsRoute` cannot carry as much information as `NavigationRoute` and the recommended migration path is to use `NavigationRouterCallback` instead of the old `RouterCallback` to request routes.
   This change might require action when integrating if you have hardcoded routes in your test suites that do not have all necessary fields to support `DirectionsRoute` to `NavigationRoute` mapping (mostly `DirectionsRoute#routeIndex`) - in these cases make sure to provide the missing data or regenerate the routes using `MapboxNavigation#requestRoutes`.
   [#5411](https://github.com/mapbox/mapbox-navigation-android/pull/5411)
```
Comment (Guardiola31337) [inline CHANGELOG.md:21]: We should add https://github.com/mapbox/mapbox-navigation-android/pull/5490 under `#### Bug fixes and improvements`.

Comment (cafesilencio) [reply CHANGELOG.md:21]: added

**Review by Zayankovsky — APPROVED** [review]

**Pull request merged**